tests/utilities/test_storage.py

```diff
@@ -89,3 +93,70 @@ def test_extract_flow_from_file_raises_on_run_register(tmpdir):
     with prefect.context({"loading_flow": True}):
         with pytest.warns(Warning):
             extract_flow_from_file(file_path=full_path)
+
+
+flow = Flow("test-module-loading")
+
+
+def test_extract_flow_from_module():
+    module_name = "tests.utilities.test_storage"
+
+    test_extract_flow_from_module.multi_level_flow = flow
+    test_extract_flow_from_module.not_a_flow = None
+    test_extract_flow_from_module.not_a_flow_factory = lambda: object()
+    test_extract_flow_from_module.invalid_callable = lambda _a, _b, **_kwargs: None
+
+    class FlowFactory:
+        @classmethod
+        def default_flow(cls):
+            return flow
+
+    test_extract_flow_from_module.callable_flow = FlowFactory.default_flow
+
+    default_flow = extract_flow_from_module(module_name)
+    attribute_flow = extract_flow_from_module(module_name, "flow")
+    module_flow = extract_flow_from_module(f"{module_name}:flow")
+    multi_level_flow = extract_flow_from_module(
+        f"{module_name}:test_extract_flow_from_module.multi_level_flow"
+    )
+    multi_level_arg_flow = extract_flow_from_module(
+        module_name, "test_extract_flow_from_module.multi_level_flow"
+    )
+    callable_flow = extract_flow_from_module(
+        f"{module_name}:test_extract_flow_from_module.callable_flow"
+    )
+
+    assert (
```

Comment:
Would you mind using different `flow` objects for each of these? Right now the logic could all be bypassed and use only `module.flow` and we'd never catch it.

Reply:
Alright, should be incorporated in https://github.com/PrefectHQ/prefect/pull/3463/commits/4d06d46ac6e8480868cbdd9064844eb8858a8339

Please let me know if there are any additional changes needed 👍